pyphare/pyphare/core/ipython.py

```diff
@@ -0,0 +1,55 @@
+
+
+
+
+import weakref
+h5_weak_refs = {}
+
+def clean_dead_h5_weak_refs():
+    for h5_path, h5_weak_ref in h5_weak_refs.copy().items():
+        if h5_weak_ref["ref"]() is None:
+            del h5_weak_refs[h5_path]
+
+# https://stackoverflow.com/a/40222538/795574
+try:
+    def exit_register(fun, *args, **kwargs):
+        def callback():
+            fun()
+            ip.events.unregister('post_execute', callback)
+        ip.events.register('post_execute', callback)
```

Comment:
probably not indented correclty?

Reply:
It's been some time since looking at this, and I'm pretty sure I had another later branch with modifications which I never added so it's quite likely